### PR TITLE
memo PictureEditDialog to stop rerenders

### LINKED ISSUE #445: ImageEditor keeps rerendering and is therefore unusable

### PR DESCRIPTION
**Pull request by olschulz**
Closes #445 

Moving your mouse constantly rerenders everything inside the FaceTaggingProvider, including the ImageEditor. The ImageEditor isn't made to be rerendered constantly and is therefore becoming unusable.

### CODE REVIEW

**Review by baerlach-git — APPROVED** [review]

File: projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureEditDialog.tsx [hunk 1]
```diff
@@ -16,15 +16,15 @@ const isDefaultCropZone = (rect: any) => {
   return rect.width < 1 || rect.height < 1;
 };
 
-const PictureEditDialog = ({
+const PictureEditDialog = memo(function PictureEditDialog({
```
Comment (MariusDoe) [inline projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureEditDialog.tsx:19]: Is there a reason you chose to convert the arrow function to a plain function?

**Review by MariusDoe — COMMENTED** [review]
Looks good. While testing, I noticed that [the code for catching a bug in tui](https://github.com/hpi-swa-lab/BP2021RH1/blob/25bc226ccc483ce1445767c4bc4407ac0144e620/projects/bp-gallery/src/components/common/editors/ImageEditor.tsx#L18-L21) doesn't work on Firefox, since the error message differs. Maybe you could change it to something like `error instanceof TypeError && error.message.includes("null")`?

File: projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureEditDialog.tsx (hunk 1 above)
Comment (olschulz) [reply projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureEditDialog.tsx:19]: getting a "Component definition is missing display name" warning if I try to use an arrow function there and decided that just using a function is probably better than adding a new eslint rule or ignore comment

**Comment by olschulz** [comment]
> Looks good. While testing, I noticed that [the code for catching a bug in tui](https://github.com/hpi-swa-lab/BP2021RH1/blob/25bc226ccc483ce1445767c4bc4407ac0144e620/projects/bp-gallery/src/components/common/editors/ImageEditor.tsx#L18-L21) doesn't work on Firefox, since the error message differs. Maybe you could change it to something like `error instanceof TypeError && error.message.includes("null")`?

The error gets caught now on firefox, but firefox just seems to ignore the event.preventDefault() and event.stopPropagation() anyways.

File: projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureEditDialog.tsx (hunk 1 above)
Comment (MariusDoe) [reply projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureEditDialog.tsx:19]: Ah, thanks!

**Review by MariusDoe — APPROVED** [review]
Huh, alright, at least we tried ;)

**Pull request merged**